lib/OpenQA/Scheduler/Scheduler.pm

```diff
@@ -393,66 +336,24 @@ sub schedule {
             catch {
                 # Again: If we see this, we are in a really bad state.
                 log_debug("Failed resetting job '$allocated->{id}' to scheduled state :( bummer! Reason: $_");
-                $failure++
-                  if $hard_busy;
             };
         }
     }
 
-    # update counters based on status
-    $failure--
-      if $failure > 0
-      && scalar(@successfully_allocated) > 0
-      && $hard_busy;
-
-    $no_actions++
-      if scalar(@successfully_allocated) == 0
-      && $hard_busy;
-
-    $no_actions = 0
-      if $no_actions > 0
-      && scalar(@successfully_allocated) > 0
-      && $hard_busy;
-
     my $elapsed_rounded = sprintf("%.5f", (time - $start_time));
     log_debug "Scheduler took ${elapsed_rounded}s to perform operations and allocated "
       . scalar(@successfully_allocated) . " jobs";
-
-    my $exceeded_timer = ((${elapsed_rounded} * 1000) > OpenQA::Scheduler::SCHEDULE_TICK_MS());
-    if ($exceeded_timer && $hard_busy) {
-        $failure += 2;
-        log_debug "Scheduling took too much time. Increasing failure count. ($failure)";
-    }
     log_debug "Allocated: " . pp($_) for @successfully_allocated;
 
-
-    my $rescheduled;
-    # Decide if we want to reschedule ourselves or not.
-    # we do that in two situations: either we had failures and CONGESTION_CONTROL is enabled,
-    # or when we take too much time to perform the operations and either CONGESTION_CONTROL or BUSY_BACKOFF is enabled
-    if (
-           ($no_actions > 0 && $hard_busy)
-        || ($failure > 0 && OpenQA::Scheduler::CONGESTION_CONTROL())
-        || (   $hard_busy
-            && $exceeded_timer))
-    {
-        my $backoff
-          = OpenQA::Scheduler::CONGESTION_CONTROL()
-          ?
-          ((2**(($failure + $no_actions) || 2)) - 1) * OpenQA::Scheduler::TIMESLOT()
-          : (((($failure + $no_actions)  || 2) / 2)**2) + OpenQA::Scheduler::SCHEDULE_TICK_MS();
-        $backoff = $backoff > OpenQA::Scheduler::MAX_BACKOFF() ? OpenQA::Scheduler::MAX_BACKOFF() : $backoff + 1000;
-        log_debug "[Congestion control] Calculated backoff: ${backoff}ms";
-        log_debug "[Congestion control] Rounds with no actions performed: ${no_actions}"
-          if $hard_busy;
-        log_debug "[Congestion control] Failures# ${failure}"
-          if $hard_busy;
-        _reschedule($backoff, 1);
-        $rescheduled++;
+    if ($summoned || $quit) {
+        log_debug("I've been summoned by the webui");
+        $summoned = 0;
+    }
+    else {
+        _reschedule(OpenQA::Scheduler::SCHEDULE_TICK_MS);
```

Comment:
The cutoff is very much wanted, but i feel to share what was the original intention here (that is, to not reschedule at a fixed value): to cope with dbus lost messages and timeouts ( because we use that also to communicate with the websocket server and scheduler is talkative expecially during allocation ) 

The ( old ?) situation was: we started to loose messages from the websocket server side (and then we lost important messages, resulting in jobs with no updates and visible as stale 'running/assigned', as noted in https://progress.opensuse.org/issues/39629 ) 

But this is sadly one behavior that could have been observed in production only, so +1 to test this in a separate branch

Reply:
But running/assigned is not handled by the scheduler at all.